src/Capability/TypeOf.hs

```diff
@@ -0,0 +1,29 @@
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE TypeInType #-}
+
+module Capability.TypeOf where
+
+-- | Type family associating a tag to the corresponding type. It is intended to
+-- simplify constraint declarations, by removing the need to redundantly specify
+-- the type associated to a tag.
+--
+-- It is poly-kinded, which allows users to define their own type of tags, and
+-- not have to declare orphan type family instances.
```

Comment:
I think a point missing from this comment is that it is _not_ recommended to define `TypeOf` with symbols. The reason is that it is not compositional and has a high likelihood to interfere with other libraries' instances of `TypeOf`. That is: symbols all belong to the same global namespace (which is partly why instances with symbols are considered orphans).

It think it needs to be made clear.

There also needs to be an example with tags made in `*`.

We could get away with `TypeOf` being defined only on `*`. But it's a nice plus that we can define it on any tag.